### ISSUE TITLE
Add rolling-update interactive cli option

### ISSUE THREAD

**Issue by mrballcb** [kubernetes/kops#4164]
**Description:**
Sometimes when doing a kops rolling-update, I would like to ssh to a new master or new node to check that some component is configured properly, has started, or just in general look around to make sure it's behaving properly.  If I know a breaking change is coming, I'd like to have kops wait after each host for me to allow it to continue with the rolling update.

**Feature Request:**
I would like to add a kops cli option, for example named `--interactive`, that would pause after each instance gets updated.  It should print out a prompt such as:
```
  Continue rolling update? (Y/n)
```
...and wait for the admin to approve before continuing with the next node.  Pressing "y" or "Y" should continue.  The default is displayed in caps (Y), so pressing Enter should also continue.  Pressing "n" or "N" could maybe print out some useful information (summary, where you stopped, etc) but it should primarily exit kops immediately and not make any more changes.

Hopeful wishlist item:  may also be able to prompt with a "(Y/n/a)" where "a" is for Always.  At that point, since the rollingClusterData is passed as a pointer, maybe can set the flag back to false and it should no longer pause and prompt after subsequent hosts.

_2. Feel free to provide a design supporting your feature request._

I have put together a simple patch that processes the commandline and calls a function after each host is updated when --interactive or -i is passed on the cli.  I chose to also use   But I don't know enough go yet to actually print out the prompt, wait for a user response, and take action based on that.

My work so far is at: https://github.com/mrballcb/kops/tree/interactive_cli_opt

Here is what it looks like running.  The line says `Pausing after finished "ip-10-114-5-151.ap-southeast-1.compute.internal"`.  It didn't actually prompt or wait or anything like that (I need help with that), but the logic is in place to call it and make it work:
```
todd $ /Users/todd/repos/go/src/k8s.io/kops/.build/local/kops rolling-update cluster louie --interactive  --yes
NAME				STATUS		NEEDUPDATE	READY	MIN	MAX	NODES
master-ap-southeast-1a-1	Ready		0		1	1	1	1
master-ap-southeast-1a-2	Ready		0		1	1	1	1
master-ap-southeast-1b-1	Ready		0		1	1	1	1
nodes				NeedsUpdate	2		0	2	3	2
I1228 11:53:54.111387   87835 instancegroups.go:135] Draining the node: "ip-10-114-5-151.ap-southeast-1.compute.internal".
node "ip-10-114-5-151.ap-southeast-1.compute.internal" cordoned
node "ip-10-114-5-151.ap-southeast-1.compute.internal" cordoned
WARNING: Deleting pods not managed by ReplicationController, ReplicaSet, Job, DaemonSet or StatefulSet: kube-proxy-ip-10-114-5-151.ap-southeast-1.compute.internal; Deleting pods with local storage: kubernetes-dashboard-2932672137-4grj0; Ignoring DaemonSet-managed pods: canal-tqfg8
pod "kubernetes-dashboard-2932672137-4grj0" evicted
node "ip-10-114-5-151.ap-southeast-1.compute.internal" drained
I1228 11:55:27.768523   87835 instancegroups.go:245] Stopping instance "i-07ac68813ef873e9b", node "ip-10-114-5-151.ap-southeast-1.compute.internal", in group "nodes.louie".
I1228 11:59:29.335102   87835 instancegroups.go:166] Validating the cluster.
I1228 11:59:32.045060   87835 instancegroups.go:217] Cluster did not validate, will try again in "30s" util duration "5m0s" expires: your kube-system pods are NOT healthy louie.
I1228 12:00:04.128262   87835 instancegroups.go:220] Cluster validated.
I1228 12:00:04.128278   87835 instancegroups.go:69] Pausing after finished "ip-10-114-5-151.ap-southeast-1.compute.internal"
I1228 12:00:04.128287   87835 instancegroups.go:135] Draining the node: "ip-10-114-5-124.ap-southeast-1.compute.internal".
node "ip-10-114-5-124.ap-southeast-1.compute.internal" cordoned
node "ip-10-114-5-124.ap-southeast-1.compute.internal" cordoned
<snip>
```

I have also verified that not passing --interactive leaves the flag set to false, and it does not enter the function which will prompt to continue.

**Comment by mrballcb** [comment]
I pushed another commit which prompts the user, accepts input, and responds based on that input. There may be better ways to do this, we'll work through them while working on a PR.

**Comment by mrballcb** [comment]
https://github.com/kubernetes/kops/pull/4166

**Comment by mrballcb** [comment]
The commit was accepted and merged, is in the pipeline for the next (kops 1.9.0) stable release.
